src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -44,6 +44,10 @@ class RsInferenceContext {
         return exprTypes[expr] ?: TyUnknown
     }
 
+    fun getExprTypeOrNull(expr: RsExpr): Ty? {
```

Comment:
The only call-site does `== null`, so perhaps `fn alreadyProcessed(expt: RsExpt): Boolan`?